include/SDL3/SDL_mutex.h

```diff
@@ -240,6 +240,209 @@ extern DECLSPEC void SDLCALL SDL_DestroyMutex(SDL_mutex * mutex);
 /* @} *//* Mutex functions */
 
 
+/**
+ *  \name Read/write lock functions
+ */
+/* @{ */
+
+/* The SDL read/write lock structure, defined in SDL_sysrwlock.c */
+struct SDL_rwlock;
+typedef struct SDL_rwlock SDL_rwlock;
+
+/*
+ *  Synchronization functions which can time out return this value
+ *  if they time out.
+ */
+#define SDL_RWLOCK_TIMEDOUT SDL_MUTEX_TIMEDOUT
+
+
+/**
+ * Create a new read/write lock.
+ *
+ * All newly-created read/write locks begin in the _unlocked_ state.
+ *
+ * Calls to SDL_LockRWLockForReading() and SDL_LockRWLockForWriting will
+ * not return while the rwlock is locked _for writing_ by another thread.
+ * See SDL_TryLockRWLockForReading() and SDL_LockRWLockForWriting() to
+ * attempt to lock without blocking.
+ *
+ * SDL read/write locks are only recursive for read-only locks!. They
+ * are not guaranteed to be fair, or provide access in a FIFO manner! They
+ * are not guaranteed to favor writers. You may not lock a rwlock for
+ * both read-only and write access at the same time from the same thread.
+ *
+ * \returns the initialized and unlocked read/write lock or NULL on
+ *          failure; call SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_DestroyRWLock
+ * \sa SDL_LockRWLockForReading
+ * \sa SDL_TryLockRWLockForReading
+ * \sa SDL_LockRWLockForWriting
+ * \sa SDL_TryLockRWLockForWriting
+ * \sa SDL_UnlockRWLock
+ */
+extern DECLSPEC SDL_rwlock *SDLCALL SDL_CreateRWLock(void);
+
+/**
+ * Lock the read/write lock for _read only_ operations.
+ *
+ * This will block until the rwlock is available, which is to say it is
+ * not locked for writing by any other thread. Of all threads waiting to
+ * lock the rwlock, all may do so at the same time as long as they are
+ * requesting read-only access; if a thread wants to lock for writing,
+ * only one may do so at a time, and no other threads, read-only or not,
+ * may hold the lock at the same time.
+ *
+ * It is legal for the owning thread to lock an already-locked rwlock
+ * for reading. It must unlock it the same number of times before it is
+ * actually made available for other threads in the system (this is known
+ * as a "recursive rwlock").
+ *
+ * SDL read/write locks are only recursive for read-only locks!. They
+ * are not guaranteed to be fair, or provide access in a FIFO manner! They
+ * are not guaranteed to favor writers. You may not lock a rwlock for
+ * both read-only and write access at the same time from the same thread.
```

Comment:
You probably don't need to duplicate this info, just put it all in the create function and then document the specific behavior for each other function.

Reply:
Fixed in ec0c774e7e4414c86543f5a9eee54a5a45372030